storage/transfermanager/option.go

```diff
@@ -69,10 +69,30 @@ func (wpt withPerOpTimeout) apply(tm *transferManagerConfig) {
 	tm.perOperationTimeout = wpt.timeout
 }
 
+// WithPartSize returns a TransferManagerOption that specifies the size of the
+// shards to transfer; that is, if the object is larger than this size, it will
+// be uploaded or downloaded in concurrent pieces.
+// The default is 32 MiB for downloads.
```

Comment:
Is this based on other langs' default?

Reply:
Yes, this is based off of Python. We should do some perf testing and change this accordingly.

Reply:
I think we should also probably set a minimum for partSize, and a way to turn off sharding (set it to zero, or negative?) But that can come in a separate PR.